### PR TITLE
Added Home Page

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, i have added carousel, card components and context.

### Files to be reviewed
- CategoryCard.jsx
- ProductCard.jsx
- Carousel.jsx
- data-context.jsx
- data-reducer.jsx

### Files to ignore
- CSS files
- header.jsx and footer.jsx
- backend `db` folder

### Please review this PR and give your feedback

### CODE REVIEW

File: src/components/card/categoryCard/CategoryCard.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <div className="card-vertical category-item">
+        <span className="category-header mt-1p5">{categoryName}</span>
+        <img className="card-img-vertical" src={url} alt={alt} loading="lazy" />
+        <span className="overlay">
+          <Link to="/" className="button button-anchor">
+            View
+          </Link>
+        </span>
+      </div>
+    </>
```
Comment (Maianki) [inline src/components/card/categoryCard/CategoryCard.jsx]: You can remove the fragments as you are already enclosing the elements in div.

File: src/components/card/productCard/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <div className="title">{title}</div>
+            <div className="card-footer-vertical">
+              <span className="discountedPrice">Rs. {discounted}</span>
+              <span className="originalPrice pl-0p5">Rs. {original}</span>
+            </div>
+          </div>
+        </Link>
+        <button className="plain-button card-button mt-0p5">Add to Cart</button>
+      </div>
+    </>
```
Comment (Maianki) [inline src/components/card/productCard/ProductCard.jsx]: Here also no need to add the fragments.

File: src/context/data-context.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          type: "SUCCESS_PRODUCTS",
+          payload: ProductsResponse.data.products,
+        });
+      } catch (error) {
+        dispatch({
+          type: "ERROR",
+          payload: "Error: Something is Wrong",
+        });
+      }
+    })();
```
Comment (Maianki) [inline src/context/data-context.jsx:37]: Just a suggestion you can create a custom hook for fetching data from api.

**Review by Maianki — APPROVED** [review]
Great work Apurva. Very clean and well-structured code.

File: src/context/data-context.jsx (hunk 3 above)
Comment (ApurvaSawant11) [reply src/context/data-context.jsx:37]: I tried it. But got a few errors.
I'll try it once again after completing basic functionalities.

File: src/components/card/categoryCard/CategoryCard.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/components/card/categoryCard/CategoryCard.jsx]: Yes...Remove it. Thanks

File: src/components/card/productCard/ProductCard.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/components/card/productCard/ProductCard.jsx]: Yes...Remove it. Thanks

File: src/components/carousel/Carousel.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    setCurrent(current === 0 ? length - 1 : current - 1);
+  };
+
+  return (
+    <>
+      <div className="carousel-container">
+        {CarouselData.map(({ id, src, alt }, index) => (
+          <div
+            className={
+              index === current ? "container display-block" : "container"
```
Comment (hsnice16) [inline src/components/carousel/Carousel.jsx:24]: Here, as class `container` is common in both conditions. So you can do something like this,
```css
className={`container ${index === current ? "display-block" : ""}`}
```

**Review by animeshgarai09 — APPROVED** [review]
Everything looks good. Modular enough for reusability. good to merge   👍🏻

File: src/components/carousel/Carousel.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+
+import { CarouselData } from "../../data/carousel.data";
+
+const Carousel = () => {
```
Comment (hsnice16) [inline src/components/carousel/Carousel.jsx:5]: this is a nice, carousel. I also wanted to implement this in mine. Got an idea from your implementation.

File: src/reducer/dataReducer.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+const initialReducerData = {
+  categoriesList: null,
+  productsList: null,
+  status: null,
+  error: null,
+};
+
+const dataReducer = (state, action) => {
```
Comment (hsnice16) [inline src/reducer/dataReducer.jsx:8]: Correct file name in the PR description. I think, it is mis-spelled. And also, as in this file we are not using any `html` tag. So I would say to name it using `.js` extension and not `.jsx`

It contains only pure javascript thing.

**Review by hsnice16 — APPROVED** [review]
Nice Work, Just added a few suggestions.

**Pull request merged**